src/sentry/workflow_engine/migration_helpers/alert_rule.py

```diff
@@ -1,20 +1,152 @@
+import logging
+
 from sentry.incidents.grouptype import MetricAlertFire
-from sentry.incidents.models.alert_rule import AlertRule
+from sentry.incidents.models.alert_rule import (
+    AlertRule,
+    AlertRuleThresholdType,
+    AlertRuleTrigger,
+    AlertRuleTriggerAction,
+)
+from sentry.integrations.services.integration import integration_service
 from sentry.snuba.models import QuerySubscription, SnubaQuery
 from sentry.users.services.user import RpcUser
 from sentry.workflow_engine.models import (
+    Action,
     AlertRuleDetector,
+    AlertRuleTriggerDataCondition,
     AlertRuleWorkflow,
+    DataCondition,
     DataConditionGroup,
+    DataConditionGroupAction,
     DataSource,
     Detector,
     DetectorState,
     DetectorWorkflow,
     Workflow,
     WorkflowDataConditionGroup,
 )
+from sentry.workflow_engine.models.data_condition import Condition
 from sentry.workflow_engine.types import DetectorPriorityLevel
 
+logger = logging.getLogger(__name__)
+
+
+def get_action_type(alert_rule_trigger_action: AlertRuleTriggerAction) -> str | None:
+    if alert_rule_trigger_action.sentry_app_id:
+        return Action.Type.SENTRY_APP
+
+    elif alert_rule_trigger_action.integration_id:
+        integration = integration_service.get_integration(
+            integration_id=alert_rule_trigger_action.integration_id
+        )
+        if not integration:
+            return None
+        for type in Action.Type.choices:
+            if type[0] == integration.provider:
+                return type[1]
+        return None
+    else:
+        return Action.Type.EMAIL
+
+
+def migrate_metric_action(
+    alert_rule_trigger_action: AlertRuleTriggerAction,
+) -> tuple[Action, DataConditionGroupAction] | None:
+    try:
+        alert_rule_trigger_data_condition = AlertRuleTriggerDataCondition.objects.get(
+            alert_rule_trigger=alert_rule_trigger_action.alert_rule_trigger
+        )
+    except AlertRuleTriggerDataCondition.DoesNotExist:
+        logger.exception(
+            "AlertRuleTriggerDataCondition does not exist",
+            extra={"alert_rule_trigger_id": alert_rule_trigger_action.alert_rule_trigger.id},
+        )
+        return None
+
+    action_type = get_action_type(alert_rule_trigger_action)
+    if not action_type:
+        logger.warning(
+            "Could not find a matching Action.Type for the trigger action",
+            extra={"alert_rule_trigger_action_id": alert_rule_trigger_action.id},
+        )
+        return None
+
+    data = {
+        "type": alert_rule_trigger_action.type,
+        "sentry_app_id": alert_rule_trigger_action.sentry_app_id,
+        "sentry_app_config": alert_rule_trigger_action.sentry_app_config,
+    }
+    action = Action.objects.create(
+        required=False,
+        type=action_type,
+        data=data,
```

Comment:
does `Action.data` need json schema validation too? it's a JSONField cc @iamrajjoshi